src/formatDate.js

```diff
@@ -42,15 +42,77 @@
  *   ['DD', 'MM', 'YYYY', '.'],
  * ) // '18.02.1997'
  *
+ *
+ *
+ *
  * @param {string} date
  * @param {string[]} fromFormat
  * @param {string[]} toFormat
  *
  * @returns {string}
  */
 
+const YEAR_FORMAT_YY = 'YY';
+const YEAR_FORMAT_YYYY = 'YYYY';
+const DAY_FORMAT = 'DD';
+const MONTH_FORMAT = 'MM';
+const PREVIOUS_CENTURY = '19';
+const CURRENT_CENTURY = '20';
+const TO_DETERMINE_CENTURY = 30;
+
+const getPrefixYear = (yearBefore, newFormat) => {
+  const yearUpdateFormat
+    = newFormat.find(partDate => partDate.includes(YEAR_FORMAT_YY));
+
+  if (yearBefore.length === yearUpdateFormat.length) {
+    return yearBefore;
+  }
+
+  if (yearBefore.toString().length > yearUpdateFormat.length) {
+    return yearBefore.slice(-2);
+  }
+
+  if (yearUpdateFormat.length === 4) {
+    const century = yearBefore < TO_DETERMINE_CENTURY
+      ? CURRENT_CENTURY
+      : PREVIOUS_CENTURY;
+
+    return century + yearBefore;
+  }
+};
```

Comment:
One minor problem with this function - it looks that there is possible situation when it returns `undefined`

Please make changes to make sure it always returns some meaningful value